pre-docs/inventory_version2.md

```diff
@@ -76,6 +76,93 @@ plan setup_lb (
 }
 ```
 
+## Creating targets in plans
+
+When using inventory v2 a new and improved API for interacting with `Target`s in inventory is available. Two new plan functions have been added: `get_target` which allows retrieving a single `Target` from inventory and `set_config` which allows setting config on a specific `Target`. The updated API also provides a way to instantiate new `Target`s with data that more closely resembles how targets are declared in an inventory file.
+
+For example, consider the following `new_targets` plan:
+```
+plan new_targets(){
+  $new_target = get_target('ssh://user:secret@1.2.3.4:2222')
+  $new_target.set_config(['ssh', 'host-key-check'], false)
+}
+```
+In the `new_targets` plan the `get_target` function returns a `Target` identified with the name `ssh://user:secret@1.2.3.4:2222`. If a `Target` with that name does not exist in inventory a new `Target` is instantiated with the `uri` and `name` attributes set to `ssh://user:secret@1.2.3.4:2222` and is added to the `all` group in inventory (where it inherits and configuration for the `all` group. If the `Target` with that name does exist, then it is simply returned. The `set_config` method is used to set a transport specific setting specified by the array of keys to that setting that matches the keys in the structured hash found in an inventoryfile under the `config` key. This illustrates how a new `Target` can be created from a URI and configuration options that are not able to be set in URI parts can be modified. 
+
+The `Target.new` method may also be used to instantiate a `Target`:
+```
+plan new_target_alternate(){
+  $config = { 'transport' => 'ssh',
+              'ssh' => {
+                'user' => 'user',
+                'password' => 'secret',
+                'host' => '1.2.3.4'
+                'port' => 2222,
+                'host-key-check' => false
+                }}
+  $new_target = Target.new('name' => 'new_target', 'config' => $config)
+}
+```
+In the `new_target_alternate` plan a new `Target` is created from a hash and added to the `all` group in inventory. If a `Target` with name `new_target` had already existed in inventory, that `Target` would be destroyed and the new `Target` would take its place.
+
+## `TargetSpec` parameters in plans
+
+When a plan parameter has the type `TargetSpec`, Bolt will ensure that values for that parameter are included in inventory. 
+
+For example:
+```
+plan auto_add(TargetSpec $nodes) {
+  return get_targets('all')
+}
+```
+The `auto_add` plan returns all of the targets in the `all` group, if the value of `$nodes` resolves to a `String` that does not match a `Target` name, a group name, a `Target` alias or a target regex, a new `Target` is created and added to the `all` group. 
+
+## `Target` reference
```

Comment:
In a better docs world I'd say this might deserve it's own page, and get linked from the inventory v2?

Reply:
@puppetlabs/bolt Open question